### DISCUSSION TITLE
How to use pulseIn in arduino

### DISCUSSION THREAD

**Discussion by zxlzy** [Rahix/avr-hal#309] [Q&A] [answered]
I want to use this https://www.arduino.cc/reference/en/language/functions/advanced-io/pulsein/

**Comment by Rahix** [comment, marked as answer]
Right now, there doesn't exist any plug-and-play replacement - you'll have to build your own.  Depending on what you need, you can repurpose some of the code from the [HC-SR04 example](https://github.com/Rahix/avr-hal/blob/main/examples/arduino-uno/src/bin/uno-hc-sr04.rs).  This part in particular uses a timer to measure the response time of he sensor:

https://github.com/Rahix/avr-hal/blob/c9b92686c019351f48ad41f093fa15a9d2f25542/examples/arduino-uno/src/bin/uno-hc-sr04.rs#L36-L69
